code/src/hooks/use-filters-with-apply/serialize-to-query-string/serializeToQueryString.test.ts

```diff
@@ -0,0 +1,50 @@
+import serializeToQueryString from "@/hooks/use-filters-with-apply/serialize-to-query-string/serializeToQueryString";
+
+describe("serializeToQueryString", () => {
+  test("serializes null correctly", () => {
+    const serializationResult = serializeToQueryString(null);
+    expect(serializationResult).toBe("null");
+  });
+
+  test("serializes 'number' type correctly", () => {
+    const serializationResult = serializeToQueryString(1);
+    expect(serializationResult).toBe("1");
+  });
+
+  test("serializes 'string' type correctly", () => {
+    const serializationResult = serializeToQueryString("hello world");
+    expect(serializationResult).toBe("hello world");
+  });
+
+  test("serializes 'boolean' type correctly", () => {
+    expect(serializeToQueryString(false)).toBe("false");
+    expect(serializeToQueryString(true)).toBe("true");
+  });
+
+  test("serializes 'Set' type correctly", () => {
+    const serializationResult = serializeToQueryString(
+      new Set(["value1", "value2"])
+    );
```

Comment:
mocked. Also in other places

Reply:
Thanks, created mocked variable for this case